### ISSUE TITLE
Strange issues with CTRL+Tab

### ISSUE THREAD

**Issue by Rendanic** [jimradford/superputty#624]
Sometime I loose the ability to switch windows with CTRL+Tab on Windows 10. I am not able to reproduce this issue, but I tried the last 4 versions of Superputty and all are affected.
I tried to register a shortcut for switching windows but that doesn't work.

This problem is extremly anoying, because this is the main reason why I use Superputty.

Is there any work arround? Am I doing anything wrong in the setup of SuperPutty?

Kind Regards

**Comment by roltd** [comment]
I have the same problem and this is not fine for me.

**Comment by Rendanic** [comment]
I found a way to reproduced the problem.
Open a couple of session. I can switch with CTRL+TAB between the session, until I click on a tab to select a session with the mouse. CTRL+TAB is not working anymore until you restart SuperPutty.
I use 1.4.0.8 on Windows 10.

**Comment by securityfun13** [comment]
I'm seeing the same or similar issue on 1.4.0.8 and 1.4.0.7 using Windows 7.  I would occasionally see this problem on my old computer, but with my new computer it happens all the time.  I agree that switching between tabs with the keyboard is one of the key features that i use SuperPutty for, so I'm very sad when it stops working.  What I see is that when I start SuperPutty the NextTab and PrevTab shortcuts are working fine (I use ctrl-pageup and ctrl-pagedown).  However, at some point they just stop working as well as all the other shortcuts.  If I close and re-open superputty the shortcuts will start working again.  However, at that point I usually have many tabs open and i don't want to close it.   Any fixes or workarounds for this would be really appreciated!  Thank you!

**Comment by sliddjur** [comment]
I am on windows server 2012 and I can reproduce the problem.

If I press ctrl+tab I successfully switch tab, and the cursor indicator is full green which indicates the tab is selected and active, right? Although when I start typing text comes up in the previous window.

**Comment by securityfun13** [comment]
@sliddjur I too have seen what you are reporting.  Often the keyboard shortcuts completely stop working, but sometimes (especially when I have a lot of tabs open), the tap switch still switches visually, but the keyboard input is still directed to the old tab ... which is pretty irritating

**Comment by Rednaz4** [comment]
Yes I can confirm I have the same issue on 1.4.0.8. It was working fine up until about 2 days ago. As others have mentioned it's a huge feature to lose.

**Comment by LiamZ** [comment]
I believe this is a duplicate of issue #483.
I have been having this issue for a long time.
I can't keep a SuperPutty session more than three days before this happens.

I think this is has to do with the handling of the focus by SuperPutty.
I feel there are specific situations of Alt-Tab from another app to SP that blasts shortcut handling.
I tried raising the log verbosity of SP to see a clue but nothing suspicious appeared.

I tried all the focus activator values, it didn't change a thing.
Happens at least from 1.4.0.4 to 1.4.0.8.

**Comment by jsmithe** [comment]
Found my way here from #521 which is also a duplicate.  After having this issue the first time I found that restarting my computer helped.  When it starts to occur restarting super putty allows me to ctrl+tab for only a few minutes. I can open a few tabs, start cycling through them, and it will stop working every time.  However, when it occurs and I restart my PC it prevents it from occurring again so quickly.  I can work out the day without it occurring.
Perhaps that info will lead to some ideas on what is causing it or it will also work for someone else and give them temporary relief from this annoyance.

**Comment by LiamZ** [comment]
I suspect this to be related to some system library version (.Net framework?).
The issue happen on one of my computers but not the other (both are windows 7 Pro x64).

There is another symptom that follow the same rule.
On the bad computer the ctrl-tab never loop when it reaches one end of the tab list.

Both are 1.4.0.8.

**Comment by PAStheLoD** [comment]
Sadly 1.4.0.9 is still affected by this. (On Win10.)

It seems that switching focus activator in the Advanced tab has no effect (or it needs a restart maybe? or all are equally affected by this bug).

**Comment by lzhang938** [comment]
Looks like it is still happening ...
It will be great it can be fixed.

**Comment by luisdvr** [comment]
I found a workaround: whenever it doesn't work use the option FocusActiveSession from Shortcuts menu (on my case set to F9):

![SuperPutty_uLsvh4zlBV](https://user-images.githubusercontent.com/3649582/105058815-5f265600-5a55-11eb-803f-2731d2d49697.png)

After using it (pressing F9) ctrl+tab works as usual

The cause is probably related to #267 or #809
